src/main/java/seedu/address/logic/parser/MarkCommandParser.java

```diff
@@ -20,19 +22,22 @@ public MarkCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args,
                 PREFIX_TUTORIAL);
 
-        Index index;
-        Tutorial tutorial;
-        try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
-            tutorial = new Tutorial(argMultimap.getValue(PREFIX_TUTORIAL).orElse(""));
-        } catch (ParseException ive) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                    MarkCommand.MESSAGE_USAGE), ive);
-        } catch (IllegalArgumentException e) {
-            throw new ParseException(e.getMessage());
+        if (!arePrefixesPresent(argMultimap, PREFIX_TUTORIAL)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
         }
 
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_TUTORIAL);
+        Index index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        Tutorial tutorial = ParserUtil.parseTutorial(argMultimap.getValue(PREFIX_TUTORIAL).get());
 
         return new MarkCommand(index, tutorial);
     }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```

Comment:
Hey Cliff, perhaps this method could be abstracted into the ArgumentMultimap class instead? As it seems like this should be a function it performs rather than the MarkCommandParser.